### PR TITLE
Install plugins

### PR DESCRIPTION
**Pull request by peterjah**
To test:

- navigate to thyra plugin manager: <http://my.massa/thyra/plugin-manager>
-Install the plugin node manager with zip or directly
- navigate to  plugin home page
- try the plugin.... (install a node ??)

### CODE REVIEW

File: web/plugin-manager/src/components/pluginBlock.tsx [hunk 1]
```diff
@@ -5,12 +5,10 @@ import { PluginProps } from "../../../shared/interfaces/IPlugin";
 import { isUp } from "../helpers/isUp";
 
 function PluginBlock(p: PluginProps) {
-    console.log("load plugin PluginBlock:",p)
-    console.log("isUp?:",isUp(p.plugin.status))
-    // Callback to set error on parent
-    function sendErrorData(errorType: string, errorMessage: string) {
-        p.setErrorData(errorType, errorMessage);
+    if(!p) {
+        return <div></div>
```
Comment (Thykof) [inline web/plugin-manager/src/components/pluginBlock.tsx:9]: according to the function definition, p can't be null

File: web/plugin-manager/src/components/installNodeManager.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
+import { useEffect, useState } from 'react';
+import axiosServices from '../services/axios';
+import { Circles } from 'react-loader-spinner'
+
+export interface InstallProps {
```
Comment (Thykof) [inline web/plugin-manager/src/components/installNodeManager.tsx:5]: duplicated code

File: web/plugin-manager/src/components/installNodeManager.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    } else if (platform.includes("linux")) {
+        return "linux";
+    }
+}
+
+function InstallNodeManager(p: InstallProps) {
+
+    const [isInstalling, setIsInstalling] = useState<boolean>(false);
+
+    const [Os, setOs] = useState<string | undefined>(undefined);
```
Comment (Thykof) [inline web/plugin-manager/src/components/installNodeManager.tsx:31]: for a variable name it should be `os` or `operatingSystem`

File: web/plugin-manager/src/components/pluginBlock.tsx [hunk 4]
```diff
@@ -5,24 +5,22 @@ import { PluginProps } from "../../../shared/interfaces/IPlugin";
...
+    async function getpluginInfo(): Promise<string | undefined> {
         try {
-            const status = await axiosServices.getpluginInfo(p.plugin.id);
-            console.log("fetchPluginStatus:",status.data)
-            setStatus(isUp(status.data))
-            return status.data
+            const res = await axiosServices.getpluginInfo(p.plugin.id);
+            setStatus(isUp(res.data.status))
+
+            return res.data?.status
```
Comment (Thykof) [inline web/plugin-manager/src/components/pluginBlock.tsx:21]: why don't you use res.data?.status also in the isUp?

**Review by Thykof — CHANGES_REQUESTED** [review]

**Review by Thykof — APPROVED** [review]

**Comment by Thykof** [comment]
I have tested:

- install manually wallet, hello world, node manager
- stop node manager in UI
- remove node manager in UI
- install node manager in UI
- see wonderful spinner
- see error in console 

```
[thyra-plugin-node-manager_darwin-amd64] Error: 2023/02/28 11:20:17 while doing register request: Post "http://my.massa/plugin-manager/register": dial tcp: lookup my.massa on 192.168.1.254:53: no such host
[thyra-plugin-node-manager_darwin-amd64] Error: panic: while doing register request: Post "http://my.massa/plugin-manager/register": dial tcp: lookup my.massa on 192.168.1.254:53: no such host


goroutine 1 [running]:
log.Panicln({0xc00029ff60?, 0xc0004030b0?, 0x205f5791a[thyra-plugin-node-manager_darwin-amd64] Error: ?})
	/opt/hostedtoolcache/go/1.19.6/x64/src/log/log.go:402 +0x65
main.main()
	/home/runner/work/thyra-node-manager-plugin/thyra-node-manager-plugin/cmd/node_manager_server/main.go:242 +0x3de
2023/02/28 11:20:17 plugin 'thyra-plugin-node-manager_darwin-amd64' exiting with error: exit status 2
```

**Comment by Thykof** [comment]
second problem I can see a plugin without name 
image

File: web/plugin-manager/src/components/installNodeManager.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    const [os, setOs] = useState<string | undefined>(undefined);
+    useEffect(() => setOs(getOperatingSystem()), []);
+
+    async function handleInstallPlugin(_: any) {
+        if (!os) {
+            p.errorHandler("error", `Unable to detect operating system`);
+            return
+        }
+        let url = "https://github.com/massalabs/thyra-node-manager-plugin/releases/latest/download/thyra-plugin-node-manager_";
+        switch (os) {
```
Comment (gregLibert) [inline web/plugin-manager/src/components/installNodeManager.tsx:40]: No default value ?

**Review by gregLibert — COMMENTED** [review]
:question: 
I can't test, I don't know what is the url for the plugin manager page

**Comment by Thykof** [comment]
> ❓ I can't test, I don't know what is the url for the plugin manager page

I have updated the PR description with the steps, tell me if you need more details

File: web/plugin-manager/src/components/installNodeManager.tsx (hunk 5 above)
Comment (Thykof) [reply web/plugin-manager/src/components/installNodeManager.tsx:40]: good point, os could be undefined, in that case should we display a flash message or do nothing? @SlnPons  what to do when the OS could not be detected?

Comment (peterjah) [reply web/plugin-manager/src/components/installNodeManager.tsx:40]: The lines just before this handle this case!
```
    if (!os) {
        p.errorHandler("error", `Unable to detect operating system`);
 ```
 TBH i 'm not very proud of this stuff. and i am not even sure its working on all Os... depending on this, we may change the autodetection to a drop down or something else to let the user chose his os

**Comment by SlnPons** [comment]
> To test:
> 
> * install manually wallet, hello world, node manager
> * navigate to thyra plugin manager: http://my.massa/thyra/home
> * stop node manager in UI
> * remove node manager in UI
> * install node manager in UI
> * see wonderful spinner

How do you install the plugin manually? 
Can't I test it via the front you did @peterjah ? (other ticket maybe?)

**Comment by peterjah** [comment]
> install the plugin

yes this PR provide the frontend to do the install. i remove this wrong description

**Review by gregLibert — APPROVED** [review]

**Comment by gregLibert** [comment]
> question I can't test, I don't know what is the url for the plugin manager page

It works on my side!

**Pull request merged**